Docker/docker-compose.yml

```diff
@@ -11,109 +11,136 @@ services:
             - var-run-postgresql:/var/run/postgresql
     # Once off container to run the DB migrations
     migrate:
-        build: ./nab
+        build:
+            context: ..
```

Comment:
One problem with this is that it sends the whole pynab folder to Docker when building each images, which is quite big (670M on my machine) and unnecessary in practice (only the requirements file is needed). So it would be nice to keep building from the Docker folder if possible (although the requirements file is an issue). Alternatively consider adding a `.Dockerignore` file at the root?

Reply:
Does it really send the pynab folder to Docker?
The COPY command still only copies Docker/nab/*.sh and requirements.txt…

Reply:
It does send the whole build folder to the Docker daemon, because the image is built by the daemon which may be on a remote machine. Then it copies what is specified in the Dockerfile inside the image. So the whole folder is not embedded in the image, but it's still sent over to the daemon. It can be seen when building the image:

![1](https://user-images.githubusercontent.com/762512/118328227-31828280-b506-11eb-8e16-5ba16df23821.png)

It turn an almost instant build into a minute or more affair. Not a big deal in the grand scheme of things, but still quite annoying for fast iteration.

Using a `.Dockerignore` file allows excluding files from the build folder to be sent to the daemon. Possibly something like this would work:

```
*
!requirements.txt
```

Reply:
Eventually the best may be to copy requirements.txt.